src/main/java/md/miller1995/Dealership/controllers/AuthController.java

```diff
@@ -1,36 +1,37 @@
 package md.miller1995.Dealership.controllers;
 
-import md.miller1995.Dealership.models.dto.UserAuthDTO;
-import md.miller1995.Dealership.services.users.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import md.miller1995.Dealership.models.dto.auth.UserAuthDTO;
+import md.miller1995.Dealership.models.dto.auth.UserRegisterDTO;
+import md.miller1995.Dealership.services.auth.AuthService;
+import md.miller1995.Dealership.utils.AuthResponse;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.stereotype.Controller;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.*;
 
-@Controller
-@RequestMapping("/auth")
+
+@RestController
+@RequestMapping("api/v1/auth")
 public class AuthController {
 
-    private final UserService userService;
+    private final AuthService authService;
 
     @Autowired
-    public AuthController(UserService userService) {
-        this.userService = userService;
+    public AuthController(AuthService authService) {
+        this.authService = authService;
     }
 
-    @GetMapping("/login")
-    public ResponseEntity<HttpStatus> loginPage(){
-        //TODO
-        return ResponseEntity.ok(HttpStatus.OK);
+    @PostMapping("/registration")
+    @Operation(summary = "this endpoint is used to register a new user in database")
+    public ResponseEntity<UserRegisterDTO> performRegistration(@RequestBody UserRegisterDTO userRegisterDTO){
+        authService.register(userRegisterDTO);
+        return ResponseEntity.ok(userRegisterDTO);
```

Comment:
Usually, you want to have a service method that is used with a `POST` HTTP action that returns a DTO as a result of the save to the database.
In our case, we could have `return ResponseEntity.ok( authService.register(userRegisterDTO));`
We could have a data mismatch if we return just the DTO from the request body as a response. Instead, we should return the response from the service layer.

Reply:
Done!